### PR TITLE
Add support for General instant app

### PR DESCRIPTION
**Pull request by pspraveenkr**
This PR adds support for "General" instant app. 

Summary of changes:
1. `CompatibilityChecker` and `configParamUtils` have been updated with support for General app
   - `generalConfigParams.ts` is intentionally empty in this PR. Will be progressively expanded 
2. Converted JSON files (used to store config params) to TS files (eventually JS files at build time)
    - This is to get around lack of browser support for natively loading non-JS resources (ex: JSON) (even with [import attributes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#loading_non-javascript_resources) using the "with" keyword)
3. Updated ESM build script to explicitly add JS extension to module paths in import statements
    - This enables native ES module loading in browsers during local development
4. Added new tests, and fix an existing test for Atlas app

Also see commit messages for details.

### CODE REVIEW

**Comment by pspraveenkr** [comment]
FYI - Even though I was able to run tests locally, I did see quite a bit of TS compiler errors (for `npm run build:test` command) that didn't seem straight forward to fix.

**Review by kellyhutchins — COMMENTED** [review]
Looks good at a glance but before we merge I'm going to test it out locally with a few of my apps. I'll do that today then comment back here when ok to merge.

File: src/baseClasses/ApplicationBase.ts [hunk 1]
```diff
@@ -752,10 +751,6 @@ export default class ApplicationBase {
       flowType: shouldUsePopup ? "authorization-code" : "auto",
     });
 
-    if (!info) {
```
Comment (kellyhutchins) [inline src/baseClasses/ApplicationBase.ts:755]: I'll try to refresh my memory on why we exit if info is null...

**Review by kellyhutchins — APPROVED** [review]
Tested locally and it looks good. I do see some issues with the use of __esri not related to this PR but I can address those after this one is merged. @Csmith246  can you review and approve the changes to the build?

**Review by Csmith246 — APPROVED** [review]

**Comment by pspraveenkr** [comment]
Thanks for reviewing @Csmith246 @kellyhutchins!

**Pull request merged**